sdks/java/io/cdc/src/main/java/org/apache/beam/io/cdc/DebeziumIO.java

```diff
@@ -0,0 +1,246 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.beam.io.cdc;
+
+import com.google.auto.value.AutoValue;
+import org.apache.beam.sdk.coders.Coder;
+import org.apache.beam.sdk.coders.MapCoder;
+import org.apache.beam.sdk.coders.StringUtf8Coder;
+import org.apache.beam.sdk.options.ValueProvider;
+import org.apache.beam.sdk.transforms.Create;
+import org.apache.beam.sdk.transforms.PTransform;
+import org.apache.beam.sdk.transforms.ParDo;
+import org.apache.beam.sdk.values.PBegin;
+import org.apache.beam.sdk.values.PCollection;
+import org.apache.beam.vendor.guava.v26_0_jre.com.google.common.base.Joiner;
+import org.apache.beam.vendor.guava.v26_0_jre.com.google.common.collect.Lists;
+import org.apache.kafka.connect.source.SourceConnector;
+import org.checkerframework.checker.nullness.qual.Nullable;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.Serializable;
+import java.util.HashMap;
+import java.util.Map;
+
+import static org.apache.beam.vendor.guava.v26_0_jre.com.google.common.base.Preconditions.checkArgument;
+
+public class DebeziumIO {
+    private static final Logger LOG = LoggerFactory.getLogger(DebeziumIO.class);
+
+    /**
+     * Read data from a Debezium source.
+     *
+     * @param <T> Type of the data to be read.
+     */
+    public static <T> Read<T> read(){
+        return new AutoValue_DebeziumIO_Read.Builder<T>().build();
+    }
+
+    /** Disallow construction of utility class. */
+    private DebeziumIO() {}
+
+    /** Implementation of {@link #read}. */
+    @AutoValue
+    public abstract static class Read<T> extends PTransform<PBegin, PCollection<T>> {
+
+        private static final long serialVersionUID = 1L;
+
+        abstract @Nullable ConnectorConfiguration getConnectorConfiguration();
+        abstract @Nullable SourceRecordMapper<T> getFormatFunction();
+        abstract @Nullable Coder<T> getCoder();
+        abstract Builder<T> toBuilder();
```

Comment:
let's have the JSON format function be a default (so users will not have to specify it)

See: https://github.com/google/auto/blob/master/value/userguide/builders-howto.md#default for how to set defaults

Reply:
for this format function, let's also set the coder ahead of time so that users will not need to by default

Reply:
I've added another [method](https://github.com/wizeline/beam/blob/faa8ae0a1bbf63ef08b99398f18b966e1b51f4fc/sdks/java/io/cdc/src/main/java/org/apache/beam/io/cdc/DebeziumIO.java#L62) for this Json string case. Now users should be able to use the generic version `<T> Read<T> read()` and the Json String version with the default mapper implementation `Read<String> readAsJson()`

You can check the test [here](https://github.com/wizeline/beam/blob/faa8ae0a1bbf63ef08b99398f18b966e1b51f4fc/sdks/java/io/cdc/src/test/java/org/apache/beam/io/cdc/DebeziumIOConnectorTest.java#L102)

What do you think?